### ISSUE TITLE
运行paddle.fluid.install_check.run_check()时报NCCL错误

### ISSUE THREAD

**Issue by ywlorna** [PaddlePaddle/Paddle#23410]
**- 版本、环境信息**：
1）PaddlePaddle版本：1.7.1.post107
2）GPU：两块p40，cuda9.0，cudnn7
3）系统环境：Centos7，python2.7.5

**- 安装方式：**
python -m pip install paddlepaddle-gpu

**- 运行import paddle.fluid没问题；
运行paddle.fluid.install_check.run_check()报错信息如下：**
Running Verify Paddle Program ... 
W0402 12:55:56.624688 30772 device_context.cc:237] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.1, Runtime API Version: 10.0
W0402 12:55:56.627355 30772 device_context.cc:245] device: 0, cuDNN Version: 7.0.
W0402 12:55:56.627372 30772 device_context.cc:271] WARNING: device: 0. The installed Paddle is compiled with CUDNN 7.6, but CUDNN version in your machine is 7.0, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
Your Paddle works well on SINGLE GPU or CPU.
I0402 12:55:58.561069 30772 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 2 cards are used, so 2 programs are executed in parallel.
/usr/lib64/python2.7/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
  "The following exception is not an EOF exception.")
2020-04-02 12:56:01,231-WARNING: Your Paddle has some problem with multiple GPU. This may be caused by:
 1. There is only 1 or 0 GPU visible on your Device;
 2. No.1 or No.2 GPU or both of them are occupied now
 3. Wrong installation of NVIDIA-NCCL2, please follow instruction on https://github.com/NVIDIA/nccl-tests 
 to test your NCCL, or reinstall it following https://docs.nvidia.com/deeplearning/sdk/nccl-install-guide/index.html

 Original Error is: 

--------------------------------------------
C++ Call Stacks (More useful to developers):
--------------------------------------------
0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
2   paddle::platform::NCCLContextMap::NCCLContextMap(std::vector<paddle::platform::Place, std::allocator<paddle::platform::Place> > const&, ncclUniqueId*, unsigned long, unsigned long)
3   paddle::framework::ParallelExecutor::ParallelExecutor(std::vector<paddle::platform::Place, std::allocator<paddle::platform::Place> > const&, std::vector<std::string, std::allocator<std::string> > const&, std::string const&, paddle::framework::Scope*, std::vector<paddle::framework::Scope*, std::allocator<paddle::framework::Scope*> > const&, paddle::framework::details::ExecutionStrategy const&, paddle::framework::details::BuildStrategy const&, paddle::framework::ir::Graph*)

----------------------
Error Message Summary:
----------------------
Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
  - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
  - Recommended issue content: all error stack information
  [unhandled system error] at (/paddle/paddle/fluid/platform/nccl_helper.h:113)
 
**- NCCL版本：**libnccl-2.6.4-1+cuda9.0.x86_64
在https://github.com/NVIDIA/nccl-tests上测试了nccl，是没问题的，如下：
# nThread 1 nGpus 2 minBytes 8 maxBytes 268435456 step: 2(factor) warmup iters: 5 iters: 20 validation: 1 
#
# Using devices
#   Rank  0 Pid  48339 on 5ac526b8-faa6-45fd-b33c-0fda50680877 device  0 [0x06] Tesla P40
#   Rank  1 Pid  48339 on 5ac526b8-faa6-45fd-b33c-0fda50680877 device  1 [0x0e] Tesla P40
#
#                                                     out-of-place                       in-place          
#       size         count    type   redop     time   algbw   busbw  error     time   algbw   busbw  error
#        (B)    (elements)                     (us)  (GB/s)  (GB/s)            (us)  (GB/s)  (GB/s)       
           8             2   float     sum    129.2    0.00    0.00  0e+00    128.3    0.00    0.00  0e+00
          16             4   float     sum    128.8    0.00    0.00  0e+00    128.1    0.00    0.00  0e+00
          32             8   float     sum    128.2    0.00    0.00  0e+00    127.9    0.00    0.00  0e+00
          64            16   float     sum    128.5    0.00    0.00  0e+00    15.78    0.00    0.00  0e+00
         128            32   float     sum    19.44    0.01    0.01  0e+00    49.95    0.00    0.00  0e+00
...

**Comment by heavengate** [comment]
这种一般是nccl没有正确安装和安装的版本不对，请参考Paddle安装文档 https://www.paddlepaddle.org.cn/install/quick 和NCCL安装文档 https://developer.nvidia.com/nccl 正确安装NCCL

**Comment by heavengate** [comment]
您好，请问问题解决了么~

**Comment by ywlorna** [comment]
问题已解决，非常感谢！






------------------




袁微

北京邮电大学/本科生/电子工程学院



13269477988

北京




&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
